apigw/src/shared/auth/saml.ts

```diff
@@ -2,16 +2,31 @@
 //
 // SPDX-License-Identifier: LGPL-2.1-or-later
 
+import { z } from 'zod'
 import { Profile, VerifiedCallback, VerifyWithoutRequest } from 'passport-saml'
 import { SamlUser } from '../routes/auth/saml/types'
+import { logWarn } from '../logging'
 
 export function toSamlVerifyFunction(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  schema: z.ZodObject<any>,
```

Comment:
According to [zod docs](https://zod.dev/?id=writing-generic-functions), `z.ZodTypeAny` would be a better type.